palace/fem/errorindicators.hpp

```diff
@@ -0,0 +1,84 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef PALACE_FEM_ERROR_INDICATORS_HPP
+#define PALACE_FEM_ERROR_INDICATORS_HPP
+
+#include "linalg/vector.hpp"
+#include "utils/communication.hpp"
+
+namespace palace
+{
+// Storage for error estimation results from the solve. Required in the AMR loop. This is
+// richer than the IndicatorsAndNormalization because it stores derived quantities, and a
+// communicator for use in adaptation.
+class ErrorIndicators
```

Comment:
OK, I agree with your prior comments and am starting to like the design more. I have a small suggested change here that I think will actually do a lot to make this more clear. Let's rename this to `ErrorIndicator`, since an object of type `ErrorIndicator` is really a single indicator over the mesh of the estimate. I get that an error indicator can be thought of as plural (where the singular would be for one element), but as a function over the mesh elements it is singular. Then we can take the min/max/sum etc. of that single indicator function. I think part of my confusion was that sometimes we had plural indicators and sometimes singular which didn't make sense on the first read. This class is good now.

With the above renaming, you could just have the following: `ErrorIndicator::GetLocal`, `ErrorIndicator::Min/Max/Mean/Norml2`. Do you think this would help?